src/Symfony/Bridge/Doctrine/ManagerRegistry.php

```diff
@@ -37,6 +39,32 @@ protected function getService($name)
      */
     protected function resetService($name)
     {
-        $this->container->set($name, null);
+        if (!$this->container->initialized($name)) {
+            return;
+        }
+        $manager = $this->container->get($name);
+
+        if (!$manager instanceof LazyLoadingInterface) {
+            @trigger_error(sprintf('Resetting a non-lazy manager service is deprecated since Symfony 3.2 and will thrown an exception in version 4.0. Set the "%s" service as lazy and require "symfony/proxy-manager-bridge" in your composer.json file instead.', $name));
+
+            $this->container->set($name, null);
+
+            return;
+        }
+        $manager->setProxyInitializer(\Closure::bind(
+            function (&$wrappedInstance, LazyLoadingInterface $manager) use ($name) {
+                if (isset($this->aliases[$name = strtolower($name)])) {
+                    $name = $this->aliases[$name];
+                }
+                $method = 'get'.strtr($name, $this->underscoreMap).'Service';
```

Comment:
Method maps is now used also. We already are enforcing a symfony container, see type hint on `ContainerAwareTrait::setContainer`